practitioner-service/src/main/java/uk/ac/ox/ndph/arts/practitioner_service/repository/AzureFhirRepository.java

```diff
@@ -0,0 +1,81 @@
+package uk.ac.ox.ndph.arts.practitioner_service.repository;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.hl7.fhir.instance.model.api.IBaseResource;
+import org.hl7.fhir.r4.model.Practitioner;
+import org.hl7.fhir.r4.model.Bundle;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.hl7.fhir.r4.model.Resource;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import ca.uhn.fhir.context.FhirContext;
+import ca.uhn.fhir.rest.server.exceptions.BaseServerResponseException;
+import ca.uhn.fhir.util.BundleUtil;
+import uk.ac.ox.ndph.arts.practitioner_service.exception.HttpStatusException;
+import uk.ac.ox.ndph.arts.practitioner_service.exception.RestException;
+
+/**
+ * Implement FhirRepository interface using HAPI client sdk and backed up by an Azure
+ * FHIR store.
+ */
+@Component
+class AzureFhirRepository implements FhirRepository {
+
+    final String PRACTITIONER_ENTITY_NAME = "Practitioner";
+
+    private final FhirContext fhirContext;
+    private final Logger logger = LoggerFactory.getLogger(AzureFhirRepository.class);
+
+    @Value("${fhir.uri}")
+    private String fhirUri = "";
+
+    public AzureFhirRepository() {
+        fhirContext = FhirContext.forR4();
+    }
+
+    public String savePractitioner(Practitioner practitioner) throws HttpStatusException {
```

Comment:
From the same reason, that spring resolves the exceptions itself, you don't have to declare throws. The practice of rethrowing checked exceptions as runtime as you did below is actually good, it is just recommended to avoid the method declaration as much as possible since eventually you find yourself with many methods that have the declaration and many clients that needs to add the try-catch block.